src/stringutils.c

```diff
@@ -2260,3 +2260,37 @@ unsigned long longdate_difference(unsigned long longdate1,unsigned long longdate
 	    interval++, m_month1++;
 	 return(interval);
 }
+
+char *format_commas(char *message)
+{
+   static char commas[50];
+   char tmp[50];
+   sprintf(tmp, "%s", message);
+   char *src, *dst;
+
+   src = tmp + strlen(tmp) - 1;
+   dst = commas + 50 - 1;
+
+   int len = strlen(tmp);
+   int cnt = 0;
+
+   do {
+      if ( *src <= '9' && *src >= '0') {
+         if( cnt && !(cnt % 3)) *dst-- = ',';
+         cnt++;
+      }
+      *dst-- = *src--;
+   } while (--len);
```

Comment:
What happens when you give this a zero length string?

Reply:
Added check to validate message coming into format_commas contains .. contents :)